guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -550,450 +355,397 @@ private RemoteRepository getRemoteRepoFromDeployment( final DeploymentRepository
      * @param fileFormats file formats to apply when finding files, ie., [ "jar", "kjar" ].
      * @return an collection of java.io.File with the matching files
      */
-    public List<File> listFiles( final String filters, List<String> fileFormats ) {
+    public List<File> listFiles(final String filters,
+                                List<String> fileFormats) {
         final List<String> wildcards = new ArrayList<String>();
         String wildcardPrefix = "";
 
-        if ( filters != null ) {
+        if (filters != null) {
             wildcardPrefix = "*" + filters;
         }
 
-        if ( fileFormats == null ) {
+        if (fileFormats == null) {
             fileFormats = new ArrayList<String>();
-            fileFormats.add( "jar" );
-            fileFormats.add( "kjar" );
-            fileFormats.add( "pom" );
+            fileFormats.add("jar");
+            fileFormats.add("kjar");
+            fileFormats.add("pom");
         }
 
-        for ( String fileFormat : fileFormats ) {
-            wildcards.add( wildcardPrefix + "*." + fileFormat );
+        for (String fileFormat : fileFormats) {
+            wildcards.add(wildcardPrefix + "*." + fileFormat);
         }
 
-        final List<File> files = new ArrayList<File>( getFiles( wildcards ) );
+        final List<File> files = new ArrayList<File>(getFiles(wildcards));
 
         return files;
     }
 
-    protected Collection<File> getFiles( final List<String> wildcards ) {
-        return FileUtils.listFiles( new File( M2_REPO_DIR ),
-                                    new WildcardFileFilter( wildcards,
-                                                            IOCase.INSENSITIVE ),
-                                    DirectoryFileFilter.DIRECTORY );
+    public List<Artifact> listArtifacts(final String filters,
+                                        List<String> fileFormats) {
+        final List<String> wildcards = new ArrayList<String>();
+        String wildcardPrefix = "";
+
+        if (filters != null) {
+            wildcardPrefix = "*" + filters;
+        }
+
+        if (fileFormats == null) {
+            fileFormats = new ArrayList<String>();
+            fileFormats.add("jar");
+            fileFormats.add("kjar");
+            fileFormats.add("pom");
+        }
+
+        for (String fileFormat : fileFormats) {
+            wildcards.add(wildcardPrefix + "*." + fileFormat);
+        }
+
+        final List<Artifact> files = new ArrayList<>(getArtifacts(wildcards));
+
+        return files;
+    }
+
+    protected Collection<File> getFiles(final List<String> wildcards) {
+        return this.repositories.stream()
+                .map(artifactRepository -> artifactRepository.listFiles(wildcards))
```

Comment:
This can be achieved using just single `flatMap`, but can be a matter of preference..

`.flatMap(artifactRepository -> artifactRepository.listFiles(wildcards).stream())`